src/coreclr/jit/emitarm64.cpp

```diff
@@ -7670,6 +7748,25 @@ void emitter::emitIns_R_S(instruction ins, emitAttr attr, regNumber reg1, int va
 
     assert(fmt != IF_NONE);
 
+    // This handles LDR duplicate instructions
+    if (emitComp->opts.OptimizationEnabled() && IsOptimisableLdrStr(ins, reg1, reg2, imm, size, fmt))
```

Comment:
This block is repeated 3 times. Probably move it to a seperate method?

Reply:
I have introduced the following method:
bool ReplacedLdrStr(...)
This has the functionality within the method, returning "true" if the instruction was replaced, so that the code does not carry on the emit the second ldr/ str instruction.

Reply:
Further changed to cater for registers being stored or loaded into ascending or descending locations.